tests/integration/test_api.py

```diff
@@ -381,3 +387,85 @@ def test_annotation_event(client, monkeypatch, httpserver):
             },
         )
     assert result.status_code == 200
+
+
+def test_prediction_collection_no_result(client):
+    result = client.simulate_get("/api/v1/predictions/")
+    assert result.status_code == 200
+    assert result.json == {"count": 0, "predictions": [], "status": "no_predictions"}
+
+
+def test_prediction_collection_no_filter(client):
+
+    prediction1 = PredictionFactory(value_tag="en:seeds")
+    result = client.simulate_get("/api/v1/predictions/")
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 1
+    assert data["status"] == "found"
+    prediction_data = data["predictions"]
+    assert prediction_data[0]["id"] == prediction1.id
+    assert prediction_data[0]["type"] == "category"
+    assert prediction_data[0]["value_tag"] == "en:seeds"
+
+    prediction2 = PredictionFactory(
+        value_tag="en:beers", data={"sample": 1}, type="brand"
+    )
+    result = client.simulate_get("/api/v1/predictions/")
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 2
+    assert data["status"] == "found"
+    prediction_data = data["predictions"]
+    assert prediction_data[1]["id"] == prediction2.id
+    assert prediction_data[1]["type"] == "brand"
+    assert prediction_data[1]["value_tag"] == "en:beers"
+
+
+def test_get_predictions():
+    prediction1 = PredictionFactory(
+        barcode="123", keep_types="category", value_tag="en:seeds"
+    )
+    prediction2 = PredictionFactory(
+        barcode="123", keep_types="category", value_tag="en:beers"
+    )
+    prediction3 = PredictionFactory(
+        barcode="123", keep_types="label", value_tag="en:eu-organic"
+    )
+    prediction4 = PredictionFactory(
+        barcode="456", keep_types="label", value_tag="en:eu-organic"
+    )
+
+    actual_prediction1 = get_predictions(barcode="123")
+    actual_items1 = [item.to_dict() for item in actual_prediction1]
+    assert actual_items1[0]["id"] == prediction1.id
+    assert actual_items1[0]["barcode"] == "123"
+    assert actual_items1[0]["type"] == "category"
+    assert actual_items1[0]["value_tag"] == "en:seeds"
+    assert actual_items1[1]["value_tag"] == "en:beers"
+    assert actual_items1[2]["value_tag"] == "en:eu-organic"
+    assert len(actual_items1) == 3
+
+    actual_prediction2 = get_predictions(keep_types=["brand"])
```

Comment:
It's cool to add some descriptive comment, so that readers don't have to process all lines to understand (I let you do it on other parts)

```suggestion
    # test that as we have no "brand" prediction, returned list is empty
    actual_prediction2 = get_predictions(keep_types=["brand"])
```